libopenage/gui/guisys/private/opengl_debug_logger.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017-2017 the openage authors. See copying.md for legal info.
+
+#include "opengl_debug_logger.h"
+
+#include <QOpenGLContext>
+#include <QOpenGLFunctions_4_4_Core>
+
+namespace qtsdl {
+
+gl_debug_parameters get_current_opengl_debug_parameters(const QOpenGLContext &current_source_context)  {
+	gl_debug_parameters params{};
+
+#ifndef __APPLE__
+	if (current_source_context.versionFunctions<QOpenGLFunctions_4_4_Core>())
+		if ((params.is_debug = current_source_context.format().options().testFlag(QSurfaceFormat::DebugContext))) {
+			glGetPointerv(GL_DEBUG_CALLBACK_FUNCTION, &params.callback);
+			params.synchronous = glIsEnabled(GL_DEBUG_OUTPUT_SYNCHRONOUS);
+		}
+#else
+	Q_UNUSED(current_source_context);
+#endif
+
+	return params;
+}
+
+void apply_opengl_debug_parameters(gl_debug_parameters params, QOpenGLContext &current_dest_context)  {
+#ifndef __APPLE__
+	if (params.is_debug && params.callback) {
```

Comment:
Should have cleaned this up like in that other place.

AFAIK, apple doesn't support this GL version, so there may be some problems. Maybe Qt can deal with them by itself without the ifdefs.